lib/todo.rb

```diff
@@ -1,15 +1,41 @@
 require 'cli/ui'
 require 'cli/kit'
 
+CLI::UI::StdoutRouter.enable
+
 module Todo
+  extend CLI::Kit::Autocall
+
   TOOL_NAME = 'todo'
+  ROOT      = File.expand_path('../..', __FILE__)
   LOG_FILE  = '/tmp/todo.log'
 
-  autoload :Config,       'todo/config'
-  autoload :Command,      'todo/command'
-  autoload :Commands,     'todo/commands'
-  autoload :EntryPoint,   'todo/entry_point'
-  autoload :ErrorHandler, 'todo/error_handler'
-  autoload :Executor,     'todo/executor'
-  autoload :Resolver,     'todo/resolver'
+  autoload(:EntryPoint, 'todo/entry_point')
+  autoload(:Commands,   'todo/commands')
+
+  autocall(:Config)  { CLI::Kit::Config.new(tool_name: TOOL_NAME) }
+  autocall(:Command) { CLI::Kit::BaseCommand }
+
+  autocall(:Executor) do
+    CLI::Kit::Executor.new(
+      tool_name: TOOL_NAME,
+      log_file:  LOG_FILE,
+    )
+  end
+
+  autocall(:Resolver) do
+    CLI::Kit::Resolver.new(command_registry: Todo::Commands::Registry)
+  end
+
+  autocall(:ErrorHandler) do
+    CLI::Kit::ErrorHandler.new(
+      log_file: LOG_FILE,
+      exception_reporter: nil
+    )
+  end
+
+  Abort        = CLI::Kit::Abort
```

Comment:
This is a bit :/ But also not... /shrug

Reply:
Yeah, unsure about this one. We use this extensively in `dev` but ... I think it's a useful pattern?

Reply:
Maybe we should just use `CLI::Kit::Abort` directly and not pretend like it's different?